gamemode/modules/chat/sv_chat.lua

```diff
@@ -0,0 +1,27 @@
+function RVR.canHear( listener, speaker )
+    if not listener:Alive() or not speaker:Alive() then
+        return false
+    end
+    
+    local speakerPos = speaker:GetPos()
+    local listenerPos = listener:GetPos()
+
+    if listenerPos:Distance( speakerPos ) > GAMEMODE.Config.Chat.CHAT_DISTANCE then
+        return false
+    end
+    
+    return true
+end
+
+function GM:PlayerCanSeePlayersChat( text, teamOnly, listener, speaker )
+    if teamOnly then
+        -- TODO check party system?
```

Comment:
Can use a hook on `PlayerCanSeePlayersChat` to detect this, so this TODO isn't needed